### PR TITLE
Create a user profile when no user profile found

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR adds the user profile creation flow (handler + front page with forms). I added the related integration test for this new page and also took the time to add the one for the update user profile page too.

I reworked a bit the `update-user-profile-form.tsx` to make it usable for both creation and updating of a user profile. 

For now, I implemented a redirection to `account/profile/user-profile/new` when we got a `404` error inside SWR fetch when navigating to `/account/profile` (so the redirection happens on front-side. Any better idea ?)

I fixed a spacement issue on the user profile form, and made the year dropdown in the datepicker input scrollable to improve UX. 

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Screenshots

image

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-jf63n3 Create Profile when profile returns a 404](https://app.clickup.com/t/jf63n3)

File: src/components/forms/profile/user-profile-form.tsx [hunk 1]
```diff
@@ -48,18 +49,33 @@ const UpdateUserProfileForm: React.FC<{ userProfileData?: Profile }> = ({
...
+              if ("createdUserProfile" in parsedResponse) {
+                router && router.push("/account/profile");
+                return;
+              }
 
-            throw new Error("Something went wrong");
-          });
+              throw new Error("Something went wrong");
+            });
+          } else {
```
Comment (Fuzznimp) [inline src/components/forms/profile/user-profile-form.tsx:65]: You can remove the `else`.

File: src/pages/api/profile/user-profile/index.ts [hunk 2]
```diff
@@ -58,6 +59,7 @@ const patchHandler: Handler = async (request, response) => {
       const profileClient = initProfileClient(profileAccessToken);
 
       const { data: updatedUserProfile } = await profileClient
+
         .patchProfile(userProfilePayload)
```
Comment (Fuzznimp) [inline src/pages/api/profile/user-profile/index.ts:63]:
```suggestion
      const { data: updatedUserProfile } = await profileClient
        .patchProfile(userProfilePayload)
```

**Review by Fuzznimp — COMMENTED** [review]
> For now, I implemented a redirection to account/profile/user-profile/new when we got a 404 error inside SWR fetch when navigating to /account/profile (so the redirection happens on front-side. Any better idea ?)

Seems the best solution to me 😃

**Review by Fuzznimp — APPROVED** [review]

**Pull request merged**